pep-XXXX.rst

```diff
@@ -0,0 +1,258 @@
+    "libxrender1", "libice6", "libsm6", "libgl1-mesa-glx", and "libglib2.0-0".
+
+On RPM-based systems, these libraries are provided by the packages ::
+
+    "ncurses", "libgcc", "libstdc++", "glibc", "libXext", "libXrender",
+    "libICE", "libSM", "mesa-libGL", and "glib2".
+
+This list was compiled by checking the external shared library dependencies of
+the Canopy [1]_ and Anaconda [2]_ distributions, which both include a wide array
+of the most popular Python modules and have been confirmed in practice to work
+across a wide swath of Linux systems in the wild.
+
+For dependencies on externally-provided versioned symbols in the above shared
+libraries, the following symbol versions are permitted: ::
+
+    GLIBC <= 2.5
+    CXXABI <= 3.4.8
+    GLIBCXX <= 3.4.9
+    GCC <= 4.2.0
+
+These symbol versions were determined by inspecting the latest symbol version
+provided in the libraries distributed with CentOS 5, a Linux distribution
+released in April 2007. In practice, this means that Python wheels which conform
+to this policy should function on almost any linux distribution released after
+this date.
+
+
+Compilation and Tooling
+=======================
+
+To support the compilation of wheels meeting the ``manylinux1`` standard, we
+provide initial drafts of two tools.
+
+The first is a Docker image based on CentOS 5.11, which is recommended as an
+easy to use self-contained build box for compiling  ``manylinux1`` wheels.
+Compiling on a more recently-released linux distribution will generally
+introduce dependencies on too-new versioned symbols. The image comes with a
+full compiler suite installed (``gcc``, ``g++``, and ``gfortran`` 4.8.2) as
+well as the latest releases of Python and pip.
+
+The second tool is a command line executable called ``auditwheel``. First, it
+inspects all of the ELF files inside a wheel to check for dependencies on
+versioned symbols or external shared libraries, and verifies conformance with
+the ``manylinux1`` policy. This includes the ability to add the new platform
+tag to conforming wheels.
+
+In addition, ``auditwheel`` has the ability to automatically modify wheels that
+depend on external shared libraries by copying those shared libraries from
+the system into the wheel itself, and modifying the appropriate RPATH entries
+such that these libraries will be picked up at runtime. This accomplishes a
+similar result as if the libraries had been statically linked without requiring
+changes to the build system.
+
+Neither of these tools are necessary to build wheels which conform with the
+``manylinux1`` policy. Similar results can usually be achieved by statically
+linking external dependencies and/or using certain inline assembly constructs
+to instruct the linker to prefer older symbol versions, however these tricks
+can be quite esoteric.
+
+
+Platform Detection in ``pip``
+=============================
+
+TODO How does ``pip`` detect that it's running on a ``manylinux1`` compatible
+system?
```

Comment:
(For the record, the name /etc/python/compatibility.cfg is inspired by
the /etc/python/cert-verification.cfg
in Nick's PEP 493)

On Wed, Jan 20, 2016 at 5:05 PM, Nathaniel Smith <njs@pobox.com> wrote:

> Huh, intersting question that I don't actually know the best answer to :-)
> might be something to ask distutils-sig about. (Nick Coghlan will probably
> have an informed opinion if nothing else.)
>
> On Wed, Jan 20, 2016 at 5:01 PM, Robert T. McGibbon <
> notifications@github.com> wrote:
>
>> In pep-XXXX.rst
>> <https://github.com/manylinux/manylinux/pull/3#discussion_r50346463>:
>>
>> > +such that these libraries will be picked up at runtime. This accomplishes a
>> > +similar result as if the libraries had been statically linked without requiring
>> > +changes to the build system.
>> > +
>> > +Neither of these tools are necessary to build wheels which conform with the
>> > +``manylinux1`` policy. Similar results can usually be achieved by statically
>> > +linking external dependencies and/or using certain inline assembly constructs
>> > +to instruct the linker to prefer older symbol versions, however these tricks
>> > +can be quite esoteric.
>> > +
>> > +
>> > +Platform Detection in ``pip``
>> > +=============================
>> > +
>> > +TODO How does ``pip`` detect that it's running on a ``manylinux1`` compatible
>> > +system?
>>
>> is the file really supposed to be in /etc/python/compatibility.cfg in
>> the real root /etc, even for a userspace installation of python/pip?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/manylinux/manylinux/pull/3/files#r50346463>.
>>
>
>
>
> --
> Nathaniel J. Smith -- https://vorpus.org <http://vorpus.org>
>



-- 
Nathaniel J. Smith -- https://vorpus.org <http://vorpus.org>